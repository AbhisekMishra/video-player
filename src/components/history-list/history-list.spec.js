import React from 'react';
import { shallow, configure } from 'enzyme';
import Component from './history-list';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('HistoryList', () => {
    let props;
    let getShallowComponent;

    beforeEach(() => {
        props = {
            historyArr: [],
            classes: {},
            setVideoLink: jest.fn(),
        };

        getShallowComponent = (newProps) => shallow(<Component {...props} {...newProps} />).dive();
    });

    describe('render', () => {
        it('snapshot of instance with no value in lists', () => {
            expect(getShallowComponent().instance()).toMatchSnapshot();
        });

        it('snapshot of instance with value in lists', () => {
            const newProps = {
                historyArr: ['link1', 'link2', 'link3', 'link4', 'link5'],
            };
            expect(getShallowComponent(newProps).instance()).toMatchSnapshot();
        });
    });

    describe('handleListItemClick', () => {
        it('should update the state value when called', () => {
            const newProps = {
                historyArr: ['link1', 'link2', 'link3', 'link4', 'link5'],
            };
            const wrapper = getShallowComponent(newProps).instance();
            wrapper.handleListItemClick({}, 'link1');
            expect(wrapper.state.selectedHistory).toBe('link1');
            expect(props.setVideoLink).toBeCalled();
        });
    });
});
