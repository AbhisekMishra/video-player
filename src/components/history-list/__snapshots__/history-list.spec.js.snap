// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList render snapshot of instance with no value in lists 1`] = `
HistoryList {
  "context": Object {},
  "handleListItemClick": [Function],
  "props": Object {
    "classes": Object {
      "button": "HistoryList-button-3",
      "header": "HistoryList-header-2",
      "list": "HistoryList-list-1",
    },
    "historyArr": Array [],
    "setVideoLink": [MockFunction],
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "selectedHistory": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <HistoryList
        classes={
          Object {
            "button": "HistoryList-button-3",
            "header": "HistoryList-header-2",
            "list": "HistoryList-list-1",
          }
        }
        historyArr={Array []}
        setVideoLink={[MockFunction]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": false,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;

exports[`HistoryList render snapshot of instance with value in lists 1`] = `
HistoryList {
  "context": Object {},
  "handleListItemClick": [Function],
  "props": Object {
    "classes": Object {
      "button": "HistoryList-button-3",
      "header": "HistoryList-header-2",
      "list": "HistoryList-list-1",
    },
    "historyArr": Array [
      "link1",
      "link2",
      "link3",
      "link4",
      "link5",
    ],
    "setVideoLink": [MockFunction],
  },
  "refs": Object {},
  "setState": [Function],
  "state": Object {
    "selectedHistory": "",
  },
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <HistoryList
        classes={
          Object {
            "button": "HistoryList-button-3",
            "header": "HistoryList-header-2",
            "list": "HistoryList-list-1",
          }
        }
        historyArr={
          Array [
            "link1",
            "link2",
            "link3",
            "link4",
            "link5",
          ]
        }
        setVideoLink={[MockFunction]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <WithStyles(Grid)
        alignItems="flex-end"
        container={true}
        justify="center"
        spacing={16}
      >
        <WithStyles(Grid)
          item={true}
          md={4}
          sm={11}
          xs={11}
        >
          <h3
            className="HistoryList-header-2"
          >
            Previously visited links
          </h3>
          <WithStyles(List)
            className="HistoryList-list-1"
            component="nav"
          >
            <WithStyles(ListItem)
              button={true}
              onClick={[Function]}
              selected={false}
            >
              <WithStyles(ListItemText)
                primary="link2"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              onClick={[Function]}
              selected={false}
            >
              <WithStyles(ListItemText)
                primary="link3"
              />
            </WithStyles(ListItem)>
            <WithStyles(ListItem)
              button={true}
              onClick={[Function]}
              selected={false}
            >
              <WithStyles(ListItemText)
                primary="link4"
              />
            </WithStyles(ListItem)>
          </WithStyles(List)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
